packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -155,11 +164,37 @@ const App = (props) => {
         }
     }, [baskets])
 
+    useEffect(() => {
+        const lastLoginTimeStamp = Date.parse(customer.lastLoginTime)
+        const creationTimeStamp = Date.parse(customer.creationDate)
+        const isNewCustomer = lastLoginTimeStamp - creationTimeStamp < NEW_CUSTOMER_MAX_AGE
+        const shouldMerge =
+            customerType === 'registered' &&
+            prevAuthType.current === 'guest' &&
+            !isNewCustomer &&
+            baskets?.baskets?.[0]?.productItems?.length > 0
+        if (shouldMerge) {
+            mergeBasket.mutate({
+                headers: {
+                    // This is not required since the request has no body
+                    // but CommerceAPI throws a '419 - Unsupported Media Type' error if this header is removed.
+                    'Content-Type': 'application/json'
+                },
+                parameters: {
+                    createDestinationBasket: true
+                }
+            })
+        }
+        // Update the current `authType` value.
+        prevAuthType.current = customerType
+    }, [customerType])
+
     useEffect(() => {
         // Listen for online status changes.
         watchOnlineStatus((isOnline) => {
             setIsOnline(isOnline)
         })
+        prevAuthType.current = customerType
```

Comment:
The initial value of customerType can be null, but it will turn to guest. We need to make sure the value of prevAuthType is guest